### DISCUSSION TITLE
.net 9 Azure devops pipeline what MSHosted agent to use

### DISCUSSION THREAD

**Discussion by gabsamples6** [dotnet/maui#25766] [General] [answered]
Hi,
trying to see if I can build our .net 8 app using .net 9 but I cannot see any ms hosted agent using .net 9.
Am I missing something ?

thanks

**Comment by LucaDN66** [comment, marked as answer]
Same here, it looks like .net 9 is not available yet in the 'windows-latest' hosted agent.
If you really need to support .net 9 you can install it manually adding these lines to the yaml

- task: UseDotNet@2  #This is needed until windows-latest hosted agent will support .net 9
  inputs:
    packageType: 'sdk'
    version: '9.0.100' 
    installationPath: '$(Agent.ToolsDirectory)/dotnet'

**Comment by gabsamples6** [comment]
@LucaDN66  thanks for your reply and apologies for late response on my side - "Few days off".. 

Thanks for that

**Comment by FlavioGoncalves-Cayas** [comment]
Just upgraded to .net 9 and am now facing the error 
`error XA0031: Java SDK 11.0 or above is required when using .NET 6 or higher. Download the latest JDK at: https://aka.ms/msopenjdk`
Although I have a build step that sets the java version
`- task: JavaToolInstaller@0
    displayName: 'Use Java 11'
    inputs:
      versionSpec: 11
      jdkArchitectureOption: x64
      jdkSourceOption: PreInstalled`

It worked perfectly fine with .net 8.0 but since upgrading to 9.0 I get the above mentioned error.
Did anyone of you face the same error?

**Reply by gabsamples6** [reply]
we do not have any java installer and our .net 9 pipeline in azure devops works fine with both iOS and android

**Reply by FlavioGoncalves-Cayas** [reply]
Thanks for the quick reply. Which agent are you using? Are you calling dotnet publish?

**Reply by gabsamples6** [reply]
We are using windows-2022

    - task: Bash@3
      displayName: Build Android App
      inputs:
        targetType: 'inline'
        script: |
          cd ${{ parameters.clientPath}}
          dotnet build src/${{parameters.clientAppPrefix}}.XXXApp/${{parameters.clientAppPrefix}}.XXXApp.csproj \
            -f "$(NET_VERSION_PREFIX)-android" \
            -c Release \
            -p:AndroidPackageFormat=aab \
            -p:AndroidKeyStore=True \
            -p:AndroidUseSharedRuntime="False" \
            -p:AndroidSigningKeyStore="$(androidKeyStore.secureFilePath)" \
            -p:AndroidSigningKeyPass="${{ parameters.keystorePassword }}" \
            -p:AndroidSigningKeyAlias="${{ parameters.keyAlias }}" \
            -p:AndroidSigningStorePass="${{ parameters.keyPassword }}"

**Reply by FlavioGoncalves-Cayas** [reply]
Thanks a lot. Somehow removing the JavaToolInstaller worked.
